specs/commandline/sign.md

```diff
@@ -0,0 +1,69 @@
+---
+title: "notation sign"
+description: "The notation sign command description and usage"
+keywords: "notation, sign"
+---
+# notation sign
+## Description
+Use `notation sign` to sign artifacts.
+
+If the signing artifact is a container image stored in a registry, the signature is pushed to the registry by default after signing successfully, and the digest of the container image is returned.
+## Outline
+```console
+$ notation sign --help
+Sign artifacts
+
+Usage:
+  notation sign [reference] [flags]
+
+Flags:
+      --cert-file string        signing certificate file
+  -e, --expiry duration         expire duration
+  -h, --help                    help for sign
+  -k, --key string              signing key name
+      --key-file string         signing key file
+  -l, --local                   reference is a local file
+      --media-type string       specify the media type of the manifest read from file or stdin (default "application/vnd.docker.distribution.manifest.v2+json")
+  -o, --output string           write signature to a specific path
+  -p, --password string         Password for registry operations (default from $NOTATION_PASSWORD)
+  -c, --pluginConfig string     list of comma-separated {key}={value} pairs that are passed as is to the plugin, refer plugin documentation to set appropriate values
+      --push                    push after successful signing (default true)
+  -r, --reference string        original reference
+  -u, --username string         Username for registry operations (default from $NOTATION_USERNAME)
+
+Global Flags:
+      --plain-http   Registry access via plain HTTP
+```
+## Usage
```

Comment:
I strongly recommend we add an _Example_ section or paragraph for each usage. The _Example_ for each command is currently missing in the Notation help doc, maybe we can start to prepare it from here.

Reply:
As discussed, we will not add `Example` in the spec, it will be added in the command help doc.  I will remove the line of `$notation sign --help`, which will cause confusion that it is an output of `notation sign` command.